website/addons/github/static/tests/githubNodeConfig.test.js

```diff
@@ -0,0 +1,425 @@
+/*global describe, it, expect, example, before, after, beforeEach, afterEach, mocha, sinon*/
+'use strict';
+var $ = require('jquery');
+
+var assert = require('chai').assert;
+
+var utils = require('tests/utils');
+var faker = require('faker');
+
+var githubNodeConfigVM = require('../githubNodeConfig')._githubNodeConfigViewModel;
+
+var API_BASE = '/api/v1/project/12345/github';
+var SETTINGS_URL = [API_BASE, 'settings', ''].join('/');
+var URLS = {
+    accounts: "/api/v1/settings/github/accounts/?pid=12345",
+    auth: "/oauth/connect/github/",
+    config: "/api/v1/project/12345/github/settings/",
+    create_repo: "/api/v1/project/12345/github/newrepo/",
+    deauthorize: "/api/v1/project/12345/github/user_auth/",
+    files: "/project/12345/files/",
+    importAuth: "/api/v1/project/12345/github/user_auth/",
+    repo_list: "/api/v1/project/12345/github/repos/",
+    settings: "/settings/addons/"
+};
+var makeSettingsEndpoint = function(result, urls) {
+    return {
+        method: 'GET',
+        url: SETTINGS_URL,
+        response: {
+            result: $.extend({}, {
+                urls: $.extend({}, URLS, urls)
+            }, result)
+        }
+    };
+};
+
+var noop = function() {};
+var APITestCase = function(cfg) {
+    this.description = cfg.description || '';
+    this.endpoint = cfg.endpoint || {
+        response: {
+            result: {}
+        }
+    };
+    this.expected = this.endpoint.response.result;
+    $.extend(this.expected, cfg.data || {});
+};
+APITestCase.prototype.run = function(test) {
+    var tc = this;
+    var server;
+    tc.before = () => {
+        server = utils.createServer(sinon, [tc.endpoint]);
+    };
+    tc.after = () => {
+        server.restore();
+    };
+    test(tc);
+};
+var APITestCases = function(test, cases) {
+    for (var i = 0; i < cases.length; i++) {
+        new APITestCase(cases[i]).run(test);
+    }
+};
+
+describe('githubNodeConfigViewModel', () => {
```

Comment:
@sloria, this borrows some from the patterns used in JS tests for S3. 

Having gotten more practice with JS testing since then, I have very mixed feeling about this APITestCase(s) pattern. While I still feel case-based tests are important, I think it might be much clearer and not too much verbose to make these explicit test cases. How do you feel about this code being spread to more places in our JS tests?